### PR TITLE
Adapt cleanup script to external AD server settings

### PR DESCRIPTION
**Pull request by exekias**

### CODE REVIEW

File: script/openchange_user_cleanup.py [hunk 1]
```diff
@@ -63,20 +64,60 @@ def active_openchange_users(self):
...
-    def __init__(self, imap_host="127.0.0.1", imap_port=143, dry_run=False,
-                 samba_helper=SambaOCHelper()):
-        self.imap_host = imap_host
-        self.imap_port = imap_port
+    def __init__(self, dry_run=False, samba_helper=SambaOCHelper()):
         self.dry_run = dry_run
         self.samba_helper = samba_helper
+        self.system_defaults_file = "/etc/sogo/sogo.conf"
+        self.user_defaults_file = \
+            os.path.expanduser("~sogo/GNUstep/Defaults/.GNUstepDefaults")
```
Comment (jkerihuel) [inline script/openchange_user_cleanup.py]: You expanduser ~sogo which I presume is Zentyal specifc. Can you remove this hardcoded path?

File: script/openchange_user_cleanup.py [hunk 2]
```diff
@@ -63,20 +64,60 @@ def active_openchange_users(self):
...
+                if tmp:
+                    connection_url = tmp
+
+        return connection_url
 
     def cleanup(self, username, password=""):
         print "===== IMAP cleanup ====="
-        client = imaplib.IMAP4(self.imap_host, self.imap_port)
+
+        connection_url = self._get_connection_url()
```
Comment (jkerihuel) [inline script/openchange_user_cleanup.py]: Can you instead provide a fallback mechanism to self._get_connection_url() is no imap_host/post is provided?

File: script/openchange_user_cleanup.py [hunk 3]
```diff
@@ -63,20 +64,60 @@ def active_openchange_users(self):
...
+
+        # imap[s]://127.0.0.1:143
+        m = re.search('(?P<scheme>.+)://(?P<host>.+):(?P<port>\d+)',
+                      connection_url)
+        if not m:
+            raise Exception("ERROR Unable to parse SOGoIMAPServer: %s" %
```
Comment (jkerihuel) [inline script/openchange_user_cleanup.py]: s/SOGoIMAPServer/IMAP Server/g

File: script/openchange_user_cleanup.py [hunk 4]
```diff
@@ -63,20 +64,60 @@ def active_openchange_users(self):
...
+                            connection_url)
+        group_dict = m.groupdict()
+        if group_dict['scheme'] not in ('imaps', 'imap'):
+            raise Exception("SOGoIMAPServer should start with imap[s]:// "
```
Comment (jkerihuel) [inline script/openchange_user_cleanup.py]: Same comment as above

**Comment by blaxter** [comment]
If this gets bigger I'd be inclined to refactor all sogo dependent code into `SOGoConfigHelper` class and use it in both `SOGoCleaner` and `ImapCleaner`. But anyway this is a silly little script, :+1: this changes for me.

**Pull request merged**